src/Symfony/Component/Validator/Constraints/Negative.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * @Annotation
+ * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
+ *
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class Negative extends LessThan
+{
+    public $message = 'This value should be negative.';
+
+    public function __construct($options = null)
+    {
+        if (null === $options) {
+            $options = [];
+        } elseif (!\is_array($options)) {
+            $options = [$this->getDefaultOption() => $options];
+        }
+
+        if (isset($options['propertyPath'])) {
+            throw new ConstraintDefinitionException(sprintf('The "propertyPath" option of the "%s" constraint cannot be set.', \get_class($this)));
+        }
+
+        if (isset($options['value'])) {
+            throw new ConstraintDefinitionException(sprintf('The "value" option of the "%s" constraint cannot be set.', \get_class($this)));
+        }
+
+        $options['value'] = 0;
+
+        parent::__construct($options);
```

Comment:
Would it make sense to abstract this code into a trait to be able to reuse it in the 4 constraint classes?

Reply:
I think it would make sense and would look like this: c77ef3b WDYT?

Reply:
Ping @fabpot